Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,391 @@
-			throw new RestException(403);
-		} catch (Exception e) {
-			logger.error("Exception:", e);
-			throw new RestException(405);
-		}
-
-	}
-	
+    private static final Logger logger = LoggerFactory.getLogger(ConceptCollectionRestController.class);
+
+    @Autowired
+    private IUserManager usermanager;
+
+    @Autowired
+    private IRestMessage restMessage;
+
+    @Autowired
+    private IWorkspaceCCManager workspaceCCManager;
+
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IConceptCollectionFactory conceptCollectionFactory;
+    @Autowired
+    private IConceptCollectionManager conceptControllerManager;
+
+    private IConceptCollection collection;
+    @Autowired
+    private IConceptCollectionFactory collectionFactory;
+
+    @Autowired
+    private IWSSecurityChecker checkWSSecurity;
+
+    @Autowired
+    private IRestVelocityFactory restVelocityFactory;
+
+    @Autowired
+    @Qualifier("conceptPowerURL")
+    private String conceptPowerURL;
+
+    @Autowired
+    @Qualifier("updateConceptPowerURLPath")
+    private String updateConceptPowerURLPath;
+
+    /**
+     * Rest interface for the getting list of concept collections of a user
+     * http://<<URL>:<PORT>>/quadriga/rest/conceptcollections
+     * http://localhost:8080/quadriga/rest/conceptcollections
+     * 
+     * @param model
+     *            ModelMap to handle the requests of client
+     * @param principal
+     *            {@link Principal} object to access authenticated user
+     * @param req
+     *            {@link HttpServletRequest} object to access request param
+     * @return Produces XML of list of concept collection details
+     * @throws RestException
+     */
+    @RequestMapping(value = "rest/conceptcollections", method = RequestMethod.GET, produces = "application/xml")
+    public ResponseEntity<String> listConceptCollections(ModelMap model, Principal principal, HttpServletRequest req)
+            throws RestException {
+        List<IConceptCollection> collectionsList = null;
+        VelocityEngine engine = null;
+        Template template = null;
+        try {
+            engine = restVelocityFactory.getVelocityEngine(req);
+            engine.init();
+            String userId = principal.getName();
+            collectionsList = conceptControllerManager.getCollectionsOwnedbyUser(userId);
+            template = engine.getTemplate("velocitytemplates/conceptcollections.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("list", collectionsList);
+
+            StringWriter writer = new StringWriter();
+            template.merge(context, writer);
+            return new ResponseEntity<String>(writer.toString(), HttpStatus.OK);
+        } catch (ResourceNotFoundException e) {
+            throw new RestException(404, e);
+        } catch (ParseErrorException e) {
+            throw new RestException(400, e);

```

Comment:
where is that thrown? by the velocity engine? If so, I don't think that should be a 400, which would indicate the client sent a bad request, but a 500, given that something would be wrong the template and there is nothing the client could do about it.